spec/models/sms/sms_decoder_spec.rb

```diff
@@ -531,6 +531,83 @@
     end
   end
 
+  describe "nested groups" do
+    let(:form) do
+      create_form(
+        questions: %w[integer multilevel_select_one_as_text_for_sms],
+        default_option_names: true
+      )
+    end
+
+    # QingGroup (root)
+    #   Questioning
+    #   Questioning (multilevel)
+    #   QingGroup
+    #     Questioning
+    #     Questioning
+    # TODO: add another nested group
```

Comment:
Need to add this please.

Reply:
Perhaps hold off on this until we have the assertion methods figured out, or you'll end up writing a bunch of throwaway code in the spec.